### PR TITLE
Example task extraction

### PR DESCRIPTION
**Pull request by Vasilije1990**
## Summary by CodeRabbit

- **New Features**
	- Introduced asynchronous functions for managing graph topologies and generating standardized node IDs, enhancing knowledge graph integration.
	- Added a new function to convert documents into chunks for processing, improving the handling of document data.
	- Implemented a new summarization task that extracts and stores summaries in a vector database.
	- Introduced a function for removing disconnected chunks and a method for checking updates on existing data.
	- Enhanced task definitions for transforming documents into graph representations.

- **Enhancements**
	- Improved the flexibility of graph topology management and streamlined node identification processes.
	- Enhanced pipeline capabilities with updated and more specific task implementations for better performance in document processing and chunk management.
	- Streamlined CI/CD workflow configuration for improved reliability in the testing environment.

### CODE REVIEW

File: cognee/tasks/graph_decomposition_to_graph_nodes/graph_decomposition_to_graph_nodes.py [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+
+class EntityNode(BaseModel):
+    uuid: str
+    name: str
+    type: str
+    description: str
+    created_at: datetime
+    updated_at: datetime
+
+async def graph_decomposition_to_graph_nodes_task(data_chunks: list[DocumentChunk], graph_model: Type[BaseModel], collection_name: str):
```
Comment (borisarzentar) [inline cognee/tasks/graph_decomposition_to_graph_nodes/graph_decomposition_to_graph_nodes.py:21]: `graph_decomposition_to_graph_nodes_task` -> `decompose_chunks_into_graph` or something like that.

**Review by borisarzentar — CHANGES_REQUESTED** [review]

File: cognee/api/v1/cognify/cognify_v2.py [hunk 2]
```diff
@@ -87,39 +89,40 @@ async def run_cognify_pipeline(dataset: Dataset):
...
-                from cognee.modules.topology.topology import TopologyEngine
-                topology_engine = TopologyEngine(infer=graph_config.infer_graph_topology)
-                root_node_id = await topology_engine.add_graph_topology(files = data)
-            elif graph_config.infer_graph_topology and not graph_config.infer_graph_topology:
-                from cognee.modules.topology.topology import TopologyEngine
-                topology_engine = TopologyEngine(infer=graph_config.infer_graph_topology)
-                await topology_engine.add_graph_topology(graph_config.topology_file_path)
-            elif not graph_config.graph_topology_task:
-                root_node_id = "ROOT"
+            #
```
Comment (borisarzentar) [inline cognee/api/v1/cognify/cognify_v2.py:92]: Let's delete this if we don't use it.

File: cognee/api/v1/cognify/cognify_v2.py [hunk 3]
```diff
@@ -87,39 +89,40 @@ async def run_cognify_pipeline(dataset: Dataset):
...
+            #     root_node_id = "ROOT"
 
             tasks = [
-                Task(process_documents, parent_node_id = root_node_id), # Classify documents and save them as a nodes in graph db, extract text chunks based on the document type
-                Task(establish_graph_topology, topology_model = KnowledgeGraph, task_config = { "batch_size": 10 }), # Set the graph topology for the document chunk data
-                Task(expand_knowledge_graph, graph_model = KnowledgeGraph, collection_name = "entities"), # Generate knowledge graphs from the document chunks and attach it to chunk nodes
-                Task(filter_affected_chunks, collection_name = "chunks"), # Find all affected chunks, so we don't process unchanged chunks
+                Task(document_to_ontology, root_node_id = root_node_id),
+                Task(source_documents_to_chunks, parent_node_id = root_node_id), # Classify documents and save them as a nodes in graph db, extract text chunks based on the document type
+                Task(chunk_to_graph_decomposition_task, topology_model = KnowledgeGraph, task_config = { "batch_size": 10 }), # Set the graph topology for the document chunk data
```
Comment (borisarzentar) [inline cognee/api/v1/cognify/cognify_v2.py:107]: Some tasks don't have `_task` in their name and some have. I prefer not having it.

File: cognee/infrastructure/databases/relational/sqlalchemy/SqlAlchemyAdapter.py [hunk 4]
```diff
@@ -26,49 +30,56 @@ def __init__(self, db_type: str, db_path: str, db_name: str, db_user: str, db_pa
...
             await connection.execute(text(f"CREATE TABLE IF NOT EXISTS {schema_name}.{table_name} ({', '.join(fields_query_parts)});"))
+            await connection.close()
 
     async def delete_table(self, table_name: str):
-        async with self.engine.connect() as connection:
-            await connection.execute(text(f"DROP TABLE IF EXISTS {table_name};"))
-
+        async with self.engine.begin() as connection:
+            await connection.execute(text(f"DROP TABLE IF EXISTS {table_name} CASCADE;"))
+            await connection.close()
```
Comment (borisarzentar) [inline cognee/infrastructure/databases/relational/sqlalchemy/SqlAlchemyAdapter.py:73]: Space between.

File: cognee/infrastructure/databases/relational/sqlalchemy/SqlAlchemyAdapter.py [hunk 5]
```diff
@@ -26,49 +30,56 @@ def __init__(self, db_type: str, db_path: str, db_name: str, db_user: str, db_pa
...
     async def insert_data(self, schema_name: str, table_name: str, data: list[dict]):
         columns = ", ".join(data[0].keys())
         values = ", ".join([f"({', '.join([f':{key}' for key in row.keys()])})" for row in data])
         insert_query = text(f"INSERT INTO {schema_name}.{table_name} ({columns}) VALUES {values};")
-        async with self.engine.connect() as connection:
+        async with self.engine.begin() as connection:
```
Comment (borisarzentar) [inline cognee/infrastructure/databases/relational/sqlalchemy/SqlAlchemyAdapter.py:78]: Space between

File: cognee/infrastructure/databases/relational/sqlalchemy/SqlAlchemyAdapter.py [hunk 6]
```diff
@@ -26,49 +30,56 @@ def __init__(self, db_type: str, db_path: str, db_name: str, db_user: str, db_pa
...
             await connection.execute(insert_query, data)
-
+            await connection.close()
```
Comment (borisarzentar) [inline cognee/infrastructure/databases/relational/sqlalchemy/SqlAlchemyAdapter.py:80]: Space between

**Review by borisarzentar — CHANGES_REQUESTED** [review]

File: cognee/api/v1/cognify/cognify_v2.py (hunk 3 above)
Comment (Vasilije1990) [reply cognee/api/v1/cognify/cognify_v2.py]: done, fixed

File: cognee/tasks/chunk_to_vector_graphstore/chunk_to_vector_graphstore.py [hunk 7]
```diff
@@ -0,0 +1,97 @@
+from cognee.infrastructure.databases.vector import DataPoint, get_vector_engine
+from cognee.infrastructure.databases.graph import get_graph_engine
+from cognee.modules.data.processing.chunk_types.DocumentChunk import DocumentChunk
+
+async def chunk_to_vector_graphstore_task(data_chunks: list[DocumentChunk], collection_name: str):
```
[the remark this answers is not in the document]
Comment (Vasilije1990) [reply cognee/tasks/chunk_to_vector_graphstore/chunk_to_vector_graphstore.py]: fixed

**Review by borisarzentar — APPROVED** [review]

**Pull request merged**